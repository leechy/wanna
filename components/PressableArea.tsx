// hooks
import { useThemeColor } from '@/hooks/useThemeColor';

// components
import { AccessibilityRole, Platform, Pressable } from 'react-native';

interface PressableAreaProps {
  style?: any;
  onPress: () => void;
  rippleColor?: string;
  children: React.ReactNode;
  testID?: string;
  accessible: boolean;
  accessibilityRole: AccessibilityRole;
  accessibilityLabel: string;
  accessibilityHint: string;
}

export function PressableArea({
  style,
  onPress,
  rippleColor,
  children,
  testID,
  accessible,
  accessibilityRole,
  accessibilityLabel,
  accessibilityHint,
}: PressableAreaProps) {
  const primaryColor = useThemeColor({}, 'primary');

  return (
    <Pressable
      android_ripple={{
        color: rippleColor || primaryColor + '33',
        borderless: false,
      }}
      style={({ pressed }) => [
        style,
        Platform.select({
          ios: { opacity: pressed ? 0.4 : 1 },
        }),
      ]}
      onPress={() => Platform.OS !== 'android' && onPress()}
      onPressOut={() => Platform.OS === 'android' && onPress()}
      testID={testID}
      accessible={accessible}
      accessibilityRole={accessibilityRole}
      accessibilityLabel={accessibilityLabel}
      accessibilityHint={accessibilityHint}
      hitSlop={{ top: 10, bottom: 10, left: 5, right: 5 }}
    >
      {children}
    </Pressable>
  );
}
