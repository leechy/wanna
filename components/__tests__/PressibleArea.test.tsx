import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Text } from 'react-native';
import { PressableArea } from '../PressableArea';

jest.mock('@/hooks/useThemeColor', () => ({
  useThemeColor: () => '#0000FF',
}));

describe('PressableArea', () => {
  it('renders correctly', () => {
    const { getByTestId } = render(
      <PressableArea onPress={() => {}} testID="pressable-area">
        <Text>Press me</Text>
      </PressableArea>
    );
    expect(getByTestId('pressable-area')).toBeTruthy();
  });

  it('calls onPress when pressed', () => {
    const onPressMock = jest.fn();
    const { getByTestId } = render(
      <PressableArea onPress={onPressMock} testID="pressable-area">
        <Text>Press me</Text>
      </PressableArea>
    );
    const pressableArea = getByTestId('pressable-area');

    fireEvent.press(pressableArea);
    expect(onPressMock).toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    const { toJSON } = render(
      <PressableArea onPress={() => {}} testID="pressable-area">
        <Text>Press me</Text>
      </PressableArea>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
