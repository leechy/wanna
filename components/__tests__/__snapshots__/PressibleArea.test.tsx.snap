// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PressableArea matches snapshot 1`] = `
<View
  accessibilityHint="Press me"
  accessibilityLabel="Press me"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      undefined,
      {
        "opacity": 1,
      },
    ]
  }
  testID="pressable-area"
>
  <Text>
    Press me
  </Text>
</View>
`;
